### PR TITLE
feat: allow network change during runtime

### PR DESCRIPTION
**Pull request by kjawadDeveloper**
## Description
So when you try to change the grpc channel it still uses the first one . Because of the static channel implementation in the code.
With this PR you can easily switch between mainnet and devnet.

## Checklist
- [ ] Targeted PR against correct branch.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [x] Run `dartfmt`.
- [x] Run `dartanalyzer`
- [ ] Updated the documentation.
- [x] Added an entry to the `CHANGELOG.md` file.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Comment by RiccardoM** [comment]
Can you revert all the changes to the `proto` folder? I think they are not needed to implement what you are describing here

**Comment by kjawadDeveloper** [comment]
@RiccardoM  this is what happened when I did dartfmt.

**Comment by RiccardoM** [comment]
> @RiccardoM this is what happened when I did dartfmt.

Please revert those changes and run `make format` instead. This should make sure that only source file get formatted instead of all the files

**Comment by kjawadDeveloper** [comment]
@RiccardoM  i think we need to update the pipeline because all the test are passing.

**Comment by kjawadDeveloper** [comment]
@RiccardoM sure let me do that.

**Comment by RiccardoM** [comment]
> @RiccardoM i think we need to update the pipeline because all the test are passing.

Yes, I've just merged #68 that should fix the running of the tests. If you merge the `main` branch into your branch it should be solved

**Comment by kjawadDeveloper** [comment]
@RiccardoM  script collect test is missing from the main code. Should I add it back?

**Comment by RiccardoM** [comment]
> @RiccardoM script collect test is missing from the main code. Should I add it back?

No need, I've just updated the `master` to fix it. I've already updated your branch and everything should be fine now. 
By the way, there are still some Proto files that have changed and I think should be reverted

**Comment by kjawadDeveloper** [comment]
@RiccardoM i tried to remove all of them . Looks like these files are left. What you suggest?

**Comment by RiccardoM** [comment]
> @RiccardoM i tried to remove all of them . Looks like these files are left. What you suggest?

I think you can simply revert the entire folder to the master commit. Something like this should work
```
git checkout 5e6eec52e7e0a6a123bbfe5b3aa2d7c4ace1bf97 -- lib/proto
```

**Comment by kjawadDeveloper** [comment]
@RiccardoM  done.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## Version 0.40.2+2
+- GRPC url will not be cache in the library in order to change the grpc url if needed.
+
+
```
Comment (RiccardoM) [inline CHANGELOG.md:4]:
```suggestion
## [Unreleased]
- `GRPCInfo#getChannel` now returns a new gRPC channel each time it's called. This makes it possible to easily switch between gRPC URLs if needed

```

**Review by RiccardoM — APPROVED** [review]

**Comment by RiccardoM** [comment]
Thanks for this changes!

**Pull request merged**